sympy/printing/printer.py

```diff
@@ -55,135 +61,115 @@
         |
         `-- p._print_Basic(expr)
 
-    if ._print_Rational method exists in the printer, then it is called,
-    and the result is returned back.
+    if ``._print_Rational`` method exists in the printer, then it is called,
+    and the result is returned back. Otherwise, the printer tries to call
+    ``._print_Number`` and so on.
 
-    otherwise, we proceed with trying Rational bases in the inheritance
-    order.
-
-3. As fall-back use the emptyPrinter method for the printer.
-
-    As fall-back self.emptyPrinter will be called with the expression. If
-    not defined in the Printer subclass this will be the same as str(expr).
-"""
+3. **As a fall-back use the emptyPrinter method for the printer.**
 
-from __future__ import print_function, division
+    As fall-back ``self.emptyPrinter`` will be called with the expression. If
+    not defined in the Printer subclass this will be the same as ``str(expr)``.
 
-from sympy import Basic, Add
+Example of Custom Printer and Custom Printing Method
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-from sympy.core.core import BasicMeta
+.. _printer_example:
 
-from functools import cmp_to_key
+This example shows how you can define a custom Printer and a custom printing method
+for your custom SymPy object. See the code below::
 
+    from sympy import Basic, Function, Symbol
+    from sympy.printing.printer import Printer
+    from sympy.printing.latex import print_latex
+    from sympy.core.basic import Basic
 
-class Printer(object):
-    """Generic printer
+    class MyBasic(Basic):
+        \"\"\" Our custom SymPy object. You can also subclass sympy.core.Expr
+        or sympy.core.Function. Both of them are subclasses of Basic.
+        \"\"\"
 
-    Its job is to provide infrastructure for implementing new printers easily.
+        def __init__(self, text):
+            self.text = text
 
-    Basically, if you want to implement a printer, all you have to do is:
+        def __str__(self):
+            return self.text
 
-    1. Subclass Printer.
+        def _latex(self, printer=None):
+            # This is the printmethod of LatexPrinter.
+            return \"\\\\textbf{{{}}}\".format(self.text)
 
-    2. Define Printer.printmethod in your subclass.
-       If a object has a method with that name, this method will be used
-       for printing.
+    class MyPrinter(Printer):
```

Comment:
Usually you want to subclass an existing printer, like StrPrinter or LatexPrinter.